### PR TITLE
Fix the content of NOTICE files

### LINKED ISSUE #7: Fix the NOTICE file content for shade jars
Here is some content of NOTICE file in amoro-shade-jackson-2:
```
Apache Amoro-shade
Copyright 2006-2024 The Apache Software Foundation

This product includes software developed at
The Apache Software Foundation (http://www.apache.org/).

amoro-shade-jackson-2
Copyright 2024 The Apache Software Foundation

This project includes software developed at
The Apache Software Foundation (http://www.apache.org/).
```

And there are some issues that should be fixed:

- (incubating) should added to the project name.
- The year range of copyright should be fixed.
- Some descriptions should be polished.

### PR DESCRIPTION
**Pull request by zhoujinsong**
close #7.

### CODE REVIEW

File: amoro-shade-zookeeper-3/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -1,8 +1,5 @@
 amoro-shade-zookeeper-3
-Copyright 2014-2023 The Apache Software Foundation
```
Comment (pjfanning) [inline amoro-shade-zookeeper-3/src/main/resources/META-INF/NOTICE:2]: this copyright should be based on the copyright in the zookeeper notice

I actually think we should keep the notice from the zookeper jar and not replace it with an amoro one

File: amoro-shade-thrift/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -1,8 +1,5 @@
 amoro-shade-thrift
-Copyright (C) 2006 - 2019, The Apache Software Foundation
```
Comment (pjfanning) [inline amoro-shade-thrift/src/main/resources/META-INF/NOTICE:2]: I think the copyright is right already and that the change is wrong

better still, we should just keep the notice from the actual thrift jar

**Comment by zhoujinsong** [comment]
Although version 0.7.0-incubating was released without incorporating these notice improvements, we should still continue to promote the optimization of notices.

I left some ncessarry changes:

- Add incubating into the project name.
- Fix the year range of copyright.

**Review by zhongqishang — APPROVED** [review]
LGTM.

**Pull request merged**